File: app/src/main/java/com/nptel/courses/online/diffutils/ModuleDiffCallback.java
package com.nptel.courses.online.diffutils;

import com.nptel.courses.online.entities.Playlist;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

public class ModuleDiffCallback extends DiffUtil.ItemCallback<Playlist> {

    @Override
    public boolean areItemsTheSame(@NonNull Playlist oldItem, @NonNull Playlist newItem) {
        return newItem.getId().equals(oldItem.getId());
    }

    @Override
    public boolean areContentsTheSame(@NonNull Playlist oldItem, @NonNull Playlist newItem) {
        return newItem.getTitle().equals(oldItem.getTitle()) && newItem.getDescription().equals(oldItem.getDescription());
    }
}

File: app/src/main/java/com/nptel/courses/online/diffutils/ModuleDiffCallback.kt
package com.nptel.courses.online.diffutils

import androidx.recyclerview.widget.DiffUtil
import com.nptel.courses.online.entities.Playlist

class ModuleDiffCallback : DiffUtil.ItemCallback<Playlist>() {
    override fun areItemsTheSame(oldItem: Playlist, newItem: Playlist): Boolean {
        return newItem.id == oldItem.id
    }

    override fun areContentsTheSame(oldItem: Playlist, newItem: Playlist): Boolean {
        return newItem.title == oldItem.title && newItem.description == oldItem.description
    }
}